pocs/utils/data.py

```diff
@@ -1,46 +1,142 @@
+        except Exception as e:
+            if not self.keep_going:
+                raise e
+            print('Failed to download {}: {}'.format(url, e))
+            return False
+        # The file has been downloaded to some directory. Move the file into the data folder.
+        try:
+            self.create_data_folder()
+            shutil.move(df, dest)
+            return True
+        except OSError as e:
+            if not self.keep_going:
+                raise e
+            print("Problem saving {}. Check permissions: {}".format(url, e))
+            return False
+
+    def create_data_folder(self):
+        """Creates the data folder if it does not exist."""
+        if not os.path.exists(self.data_folder):
+            print("Creating data folder: {}.".format(self.data_folder))
+            os.makedirs(self.data_folder)
+
+
+def main():
     parser = argparse.ArgumentParser(
-        description=__doc__,
-        formatter_class=argparse.RawDescriptionHelpFormatter)
+        description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument(
+        '--folder',
+        help='Destination folder for astrometry indices. Default: {}'.format(DEFAULT_DATA_FOLDER),
+        default=DEFAULT_DATA_FOLDER)
+
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument(
+        '--keep-going',
+        action='store_true',
+        help='Ignore download failures and keep going to the other downloads (default)')
+    group.add_argument(
+        '--no-keep-going', action='store_true', help='Fail immediately if any download fails')
+
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument(
+        '--narrow-field', action='store_true', help='Do download narrow field indices')
+    group.add_argument(
+        '--no-narrow-field',
+        action='store_true',
+        help='Skip downloading narrow field indices (default)')
 
-    parser.add_argument('--folder', help='Folder to place astrometry data')
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument(
+        '--wide-field', action='store_true', help='Do download wide field indices (default)')
+    group.add_argument(
+        '--no-wide-field', action='store_true', help='Skip downloading wide field indices')
 
     args = parser.parse_args()
 
     if args.folder and not os.path.exists(args.folder):
-        print("{} does not exist.".format(args.folder))
+        print("Warning, data folder {} does not exist.".format(args.folder))
 
-    download_all_files(data_folder=args.folder)
+    # --no_narrow_field is the default, so the the args list below ignores args.no_narrow_field.
+    dl = Downloader(
+        data_folder=args.folder,
+        keep_going=args.keep_going or not args.no_keep_going,
+        narrow_field=args.narrow_field,
+        wide_field=args.wide_field or not args.no_wide_field)
+    return dl.download_all_files()
+
+
+if __name__ == '__main__':
```

Comment:
I want to say that the argparse statements should be in this `__name__` and that those arguments should then be passed to a `main` but I can't find anything to support why that would be a preferred way.

Reply:
I don't see a need for that... except that I didn't pass in `sys.argv[1:]` so that it could be passed to `argparse.parse()`. However, the examples I found didn't bother with that. And it wouldn't make sense to call the method `main` if the parsing had already occurred Anyway, we can fix that later if it makes sense.